### PR TITLE
Issue #45: Refactor to match changed catalog Queue inteface signatures

### PR DESCRIPTION
**Pull request by Vinai**
Reference https://github.com/lizards-and-pumpkins/catalog/issues/891
Reference https://github.com/lizards-and-pumpkins/catalog/pull/892

### CODE REVIEW

**Comment by Vinai** [comment]
The build will pass once https://github.com/lizards-and-pumpkins/catalog/pull/892 is merged

**Comment by tim-bezhashvyly** [comment]
lizards-and-pumpkins/catalog#892 is merged but now the PR has to be upstreamed/rebased, (unfortunately) PHP 5.6 compatibility has to be restored.

**Comment by Vinai** [comment]
Depends on catalog dev-master now so the build passes. Once there is a release with the new interface signature the catalog dependency version should be adjusted again.

File: composer.json [hunk 1]
```diff
@@ -17,17 +17,21 @@
         }
     ],
     "require": {
-        "php": "5.6.*|7.0.*",
-        "lizards-and-pumpkins/catalog": "^0.6.0-alpha"
+        "php": "^5.6|^7.0",
```
Comment (tim-bezhashvyly) [inline composer.json]: Sorry for bugging again but this either has to be `5.6.*|7.0.*` or `^5.6.x|^7.0.x` if minimum "bugfix" version must be defined.

File: src/FileQueue.php [hunk 2]
```diff
@@ -139,40 +137,12 @@ private function getNextFile()
...
     /**
-     * @param mixed $data
-     * @return string
-     */
-    private function serialize($data)
-    {
-        try {
-            return serialize($data);
-        } catch (\Exception $e) {
-            throw new NotSerializableException($e->getMessage());
```
Comment (tim-bezhashvyly) [inline src/FileQueue.php]: I believe the exception class can be dropped. Also you might now have a stray import.

File: src/FileQueue.php [hunk 3]
```diff
@@ -139,40 +137,12 @@ private function getNextFile()
...
-        return ((string) microtime(true) * 10000) . $classNameSuffix;
-    }
-
-    /**
-     * @param string $className
+     * @param Message $data
      * @return string
      */
-    private function getBaseClassName($className)
+    protected function getFileNameForMessage(Message $data)
```
Comment (tim-bezhashvyly) [inline src/FileQueue.php]: Shouldn't it be private?

File: tests/Unit/Suites/FileQueueTest.php [hunk 4]
```diff
@@ -33,6 +33,15 @@ private function createFileQueueInstance()
         return new FileQueue($this->storagePath, $this->lockFilePath);
     }
 
+    /**
+     * @param string $name
+     * @return Message
+     */
+    private function createTestMessage($name = 'dummy')
```
Comment (tim-bezhashvyly) [inline tests/Unit/Suites/FileQueueTest.php]: I know I sound like Lucius Malfoy with his pure blood stuff but I just can't pass by an optional parameter even this one will (probably) do no harm.

**Comment by tim-bezhashvyly** [comment]
Review is done.

File: src/FileQueue.php (hunk 2 above)
Comment (Vinai) [inline src/FileQueue.php]: The exception class is already gone from the catalog repository.
Import removed.

File: src/FileQueue.php (hunk 3 above)
Comment (Vinai) [inline src/FileQueue.php]: No, it is overridden by `\LizardsAndPumpkins\Messaging\Queue\File\FileNameFixtureFileQueue::getFileNameForMessage()`

Comment (tim-bezhashvyly) [inline src/FileQueue.php]: Then maybe final?

Comment (tim-bezhashvyly) [inline src/FileQueue.php]: Ah, sorry. My bad.

Comment (tim-bezhashvyly) [inline src/FileQueue.php]: But still doesn't look nice.

File: tests/Unit/Suites/FileQueueTest.php (hunk 4 above)
Comment (Vinai) [inline tests/Unit/Suites/FileQueueTest.php]: Refactored.

File: composer.json (hunk 1 above)
Comment (Vinai) [inline composer.json]: Fixed

**Comment by Vinai** [comment]
Ready for next review

**Pull request merged**